### PR TITLE
feature: generic entity viewer

### PR DESCRIPTION
**Pull request by soofstad**
## What does this pull request change?
- Creates a new core component `EntityView` meant to be the single entry point for rendering entities (replaces UIPluginSelector)
- Update dm-core-plugins to be compatible with latest DMSS API changes

## Why is this pull request needed?
- There should be one simple to use component to render _a single recipe_ of an entity

## Issues related to this change
none

### CODE REVIEW

File: packages/dm-core/src/components/EntityView.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export const EntityView = (props: IUIPlugin): JSX.Element => {
+  const { idReference, type, onSubmit, onOpen } = props
+  const { recipe, isLoading, error, getUiPlugin } = useRecipe(type)
+
+  if (isLoading)
+    return (
+      <div style={{ alignSelf: 'center', padding: '50px' }}>
+        <CircularProgress color="primary" />
+      </div>
+    )
```
Comment (KristianKjerstad) [inline packages/dm-core/src/components/EntityView.tsx:24]: Could use the Loading.tsx component instead

File: packages/dm-core/src/components/ViewCreator/ReferenceView.tsx [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
 export const ReferenceView = (props: TReferenceView) => {
-  const { absoluteDottedId, type, viewConfig, getUiPlugin } = props
-  const { recipe, isLoading: isRecipeLoading, error } = useRecipe(
+  const { absoluteDottedId, type, viewConfig } = props
+  const { recipe, isLoading, error, getUiPlugin } = useRecipe(
     type,
     viewConfig.recipe
   )
-  if (isRecipeLoading) return <>Loading...</>
+  if (isLoading) return <>Loading...</>
```
Comment (KristianKjerstad) [inline packages/dm-core/src/components/ViewCreator/ReferenceView.tsx:17]: Could use Loading component from Loading.tsx

**Review by KristianKjerstad — APPROVED** [review]
good!

Just a few small suggestions to consider

File: packages/dm-core/src/components/EntityView.tsx (hunk 1 above)
Comment (soofstad) [reply packages/dm-core/src/components/EntityView.tsx:26]: fixed

File: packages/dm-core/src/components/ViewCreator/ReferenceView.tsx (hunk 2 above)
Comment (soofstad) [reply packages/dm-core/src/components/ViewCreator/ReferenceView.tsx]: fixed

File: packages/dm-core/src/hooks/useRecipe.tsx [hunk 3]
```diff
@@ -62,12 +69,14 @@ interface IUseRecipe {
...
 export const useRecipe = (typeRef: string, recipeName?: string): IUseRecipe => {
   const { initialUiRecipe, uiRecipes, isLoading, error } = useBlueprint(typeRef)
+  const { loading, getUiPlugin } = useContext(UiPluginContext)
 
   return {
     recipe: isLoading
       ? undefined
       : findRecipe(initialUiRecipe, uiRecipes, recipeName),
-    isLoading,
+    isLoading: isLoading && loading,
```
[the remark this answers is not in the document]
Comment (soofstad) [reply packages/dm-core/src/hooks/useRecipe.tsx:78]: fixed

**Review by eoaksnes — APPROVED** [review]
Nice work!

**Pull request merged**